kcidb/orm/__init__.py

```diff
@@ -3,1287 +3,27 @@
-                origin=_TEST['origin'],
-                path=_TEST['path'],
-                environment_comment=_TEST_ENVIRONMENT['comment'],
-                environment_misc=_TEST_ENVIRONMENT['misc'],
-                status=_TEST['status'],
-                waived=_TEST['waived'],
-                start_time=_TEST['start_time'],
-                duration=_TEST['duration'],
-                output_files=_TEST['output_files'],
-                log_url=_TEST['log_url'],
-                log_excerpt=_TEST['log_excerpt'],
-                comment=_TEST['comment'],
-                misc=_TEST['misc'],
-            ),
-            required_fields={'id', 'origin', 'build_id'},
-            id_fields=("id",),
-            children=dict(
-                incident=("test_id",),
-            ),
-        ),
-        bug=dict(
-            field_json_schemas=dict(
-                url=_ISSUE['report_url'],
-                subject=_ISSUE['report_subject'],
-                culprit_code=_ISSUE_CULPRIT['code'],
-                culprit_tool=_ISSUE_CULPRIT['tool'],
-                culprit_harness=_ISSUE_CULPRIT['harness'],
-            ),
-            required_fields={'url'},
-            id_fields=("url",),
-            children=dict(
-                issue=("report_url",),
-            ),
-        ),
-        issue=dict(
-            field_json_schemas=dict(
-                id=_ISSUE['id'],
-                version=_ISSUE['version'],
-                origin=_ISSUE['origin'],
-                report_url=_ISSUE['report_url'],
-                report_subject=_ISSUE['report_subject'],
-                culprit_code=_ISSUE_CULPRIT['code'],
-                culprit_tool=_ISSUE_CULPRIT['tool'],
-                culprit_harness=_ISSUE_CULPRIT['harness'],
-                build_valid=_ISSUE['build_valid'],
-                test_status=_ISSUE['test_status'],
-                comment=_ISSUE['comment'],
-                misc=_ISSUE['misc'],
-            ),
-            required_fields={'id', 'version', 'origin'},
-            id_fields=("id",),
-            children=dict(
-                incident=("issue_id",),
-            ),
-        ),
-        incident=dict(
-            field_json_schemas=dict(
-                id=_INCIDENT['id'],
-                origin=_INCIDENT['origin'],
-                issue_id=_INCIDENT['issue_id'],
-                issue_version=_INCIDENT['issue_version'],
-                build_id=_INCIDENT['build_id'],
-                test_id=_INCIDENT['test_id'],
-                comment=_INCIDENT['comment'],
-                misc=_INCIDENT['misc'],
-            ),
-            required_fields={'id', 'origin', 'issue_id', 'issue_version'},
-            id_fields=("id",),
-        ),
-    )
-)
-
 assert all(k.endswith("s") for k in io.SCHEMA.graph if k), \
     "Not all I/O object list names end with 's'"
 
 assert set(SCHEMA.types) >= \
     set(k[:-1] for k in io.SCHEMA.graph if k), \
     "OO types are not a superset of I/O types"
 
```

Comment:
These assertions should go into the `data` submodule, together with the `SCHEMA`, as they're closely tied to it, and are verifying either its pre-conditions, or the `SCHEMA` itself.

Reply:
done!!!